rmgpy/species.py

```diff
@@ -365,8 +365,16 @@ def from_adjacency_list(self, adjlist, raise_atomtype_exception=True, raise_char
         list in the `molecule` attribute. Does not generate resonance isomers
         of the loaded molecule.
         """
+        lines = adjlist.splitlines()
+
+        if len(lines[0].split()) == 1:
+            label = lines.pop(0) # remove the first line if it is a label before detecting cutting label
+
+        adjlist_no_label = '\n'.join(lines)
```

Comment:
if the first line *isn't* a label, then you don't need to do the whole `'\n'.join(lines)` operation. That could be inside the `if` block, and an `else` block is just `adjlist_no_label = adjlist`